### PR TITLE
Apache and Ngnix OtelExporterOtlpHeaders changes 

### PR DESCRIPTION
**Pull request by gnm444**
Apache and Ngnix OtelExporterHeaders changes related to https://github.com/open-telemetry/opentelemetry-cpp-contrib/issues/198

I have tested this using Wireshark. Please find below the headers information captured using tcpdump

```
..$.....................@.....@...... .................?..
04:38:22.473273 IP 172.18.0.4.4317 > 172.18.0.2.49424: tcp 0
E..4..@.@.?..............z)/.._y....XQ.....
..<Mo. .
04:38:22.480960 IP 172.18.0.4.4317 > 172.18.0.2.49424: tcp 15
E..C..@.@.?..............z)/.._y....X`.....
..<Uo. ..............@.
04:38:22.481068 IP 172.18.0.2.49424 > 172.18.0.4.4317: tcp 0
E..4..@.@.3..............._y.z)>....XQ.....
o. ...<U
04:38:22.482460 IP 172.18.0.2.49424 > 172.18.0.4.4317: tcp 1357
E.....@.@..o.............._y.z)>....]......
o. ...<U...........5........@
:authority.collector:4317@.:path;/opentelemetry.proto.collector.trace.v1.TraceService/Export..**api-key.abc123**@.te.trailers@.content-type.application/grpc@
user-agent-grpc-c++/1.36.4 grpc-c/15.0.0 (linux; chttp2)@.grpc-accept-encoding.identity,deflate,gzip@.accept-encoding.identity,gzip@.grpc-timeout.9970m...........................
..
..
 
.telemetry.sdk.version..
.1.2.0
.
.telemetry.sdk.language..
.cpp
+
.service.namespace..
.DemoServiceNamespace
.
.service.name.
.DemoService
%
.telemetry.sdk.name..
.opentelemetry
'
.service.instance.id..
.DemoInstanceId
.
.webengine.name..
.Apache...
```

### CODE REVIEW

File: instrumentation/otel-webserver-module/opentelemetry_module.conf [hunk 1]
```diff
@@ -14,6 +14,7 @@ ApacheModuleEnabled ON
 #ApacheModule Otel Exporter details
 ApacheModuleOtelSpanExporter otlp
 ApacheModuleOtelExporterEndpoint collector:4317
+ApacheModuleOtelExporterHeaders api-key=abc123
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/opentelemetry_module.conf:17]: Lets keep this commented in the final PR, since the value is for testing

Comment (gnm444) [reply instrumentation/otel-webserver-module/opentelemetry_module.conf:17]: Let's discuss what is the format of this ApacheModuleOtelExporterHeaders, should we process by delimiters @DebajitDas @kpratyus 

I just realized the right guide line is "Specifying headers via environment variables"
The OTEL_EXPORTER_OTLP_HEADERS, OTEL_EXPORTER_OTLP_TRACES_HEADERS, OTEL_EXPORTER_OTLP_METRICS_HEADERS environment variables will contain a list of key value pairs, and these are expected to be represented in a format matching to the [W3C Correlation-Context](https://github.com/w3c/baggage/blob/master/baggage/HTTP_HEADER_FORMAT.md), except that additional semi-colon delimited metadata is not supported, i.e.: key1=value1,key2=value2. All attribute values MUST be considered strings.

File: instrumentation/otel-webserver-module/include/core/api/TenantConfig.h [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 #include <string>
 #include <iostream>
+#include <map>
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/include/core/api/TenantConfig.h:20]: Is it required here?

File: instrumentation/otel-webserver-module/include/core/api/AppdynamicsSdk.h [hunk 3]
```diff
@@ -27,6 +27,7 @@
 /* {{{ For API user: environment variables to specify Controller connectivity */
 #define OTEL_SDK_ENV_OTEL_EXPORTER_TYPE "OTEL_SDK_ENV_OTEL_EXPORTER_TYPE"
 #define OTEL_SDK_ENV_OTEL_EXPORTER_ENDPOINT "OTEL_SDK_ENV_OTEL_EXPORTER_ENDPOINT"             /*required*/
+#define OTEL_SDK_ENV_OTEL_EXPORTER_OTELHEADERS "OTEL_SDK_ENV_OTEL_EXPORTER_OTELHEADERS"             /*optional*/
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/include/core/api/AppdynamicsSdk.h:30]: Can we have a better name here? to many OTEL
may be something below
OTEL_SDK_ENV_OTEL_EXPORTER_OTLPHEADERS

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c [hunk 4]
```diff
@@ -202,6 +202,13 @@ static ngx_command_t ngx_http_opentelemetry_commands[] = {
       offsetof(ngx_http_opentelemetry_loc_conf_t, nginxModuleOtelExporterEndpoint),
       NULL},
 
+    { ngx_string("NginxModuleOtelExporterOtelHeaders"),
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:205]: should be NginxModuleOtelExporterHeaders ? to be consisten with Apache

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c [hunk 5]
```diff
@@ -202,6 +202,13 @@ static ngx_command_t ngx_http_opentelemetry_commands[] = {
...
+      NGX_HTTP_MAIN_CONF|NGX_HTTP_SRV_CONF|NGX_HTTP_LOC_CONF|NGX_CONF_TAKE1,
+      ngx_conf_set_str_slot,
+      NGX_HTTP_LOC_CONF_OFFSET,
+      offsetof(ngx_http_opentelemetry_loc_conf_t, nginxModuleOtelExporterOtelHeaders),
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:209]: Same as above in naming

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c [hunk 6]
```diff
@@ -1447,6 +1460,7 @@ static void traceConfig(ngx_http_request_t *r, ngx_http_opentelemetry_loc_conf_t
     ngx_writeTrace(r->connection->log, __func__, " Config { :"
                                                       "(Enabled=\"%ld\")"
                                                       "(OtelExporterEndpoint=\"%s\")"
+                                                      "(OtelExporterOtelHeader=\"%s\")"
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:1463]: OtlpHeader

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c [hunk 7]
```diff
@@ -1469,6 +1483,7 @@ static void traceConfig(ngx_http_request_t *r, ngx_http_opentelemetry_loc_conf_t
                                                       " }",
                                                       conf->nginxModuleEnabled,
                                                       (conf->nginxModuleOtelExporterEndpoint).data,
+                                                      (conf->nginxModuleOtelExporterOtelHeaders).data,
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.c:1486]: OtlpHeaders

File: instrumentation/otel-webserver-module/include/core/api/TenantConfig.h [hunk 8]
```diff
@@ -61,6 +63,7 @@ class TenantConfig
     void setServiceInstanceId(const std::string& serviceInstanceId) { this->serviceInstanceId = serviceInstanceId; }
     void setOtelExporterType(const std::string& otelExporterType) { this->otelExporterType = otelExporterType; }
     void setOtelExporterEndpoint(const std::string& otelExporterEndpoint) { this->otelExporterEndpoint = otelExporterEndpoint; }
+    void setOtelExporterHeaders(const std::string& otelExporterHeaders) { this->otelExporterHeaders = otelExporterHeaders; }
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/include/core/api/TenantConfig.h:66]: This function is not called anywhere. You need to handle in ApiUtils.cpp

File: instrumentation/otel-webserver-module/src/apache/ApacheConfig.cpp [hunk 9]
```diff
@@ -771,6 +779,9 @@ otel_cfg* ApacheConfigHandlers::getProcessConfig(const request_rec* r)
     process_cfg->otelExporterEndpoint = apr_pstrdup(r->server->process->pool, our_config->otelExporterEndpoint);
     process_cfg->otelExporterEndpoint_initialized = our_config->otelExporterEndpoint_initialized;
 
+    process_cfg->otelExporterOtlpHeaders = apr_pstrdup(r->server->process->pool, our_config->otelExporterEndpoint);
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/src/apache/ApacheConfig.cpp:782]: our_config->otelExporterEndpoint ??? should be otlpheaders

File: instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.h [hunk 10]
```diff
@@ -101,6 +102,7 @@ typedef struct {
     ngx_str_t   nginxModuleSegmentParameter;
     ngx_str_t   nginxModuleRequestHeaders;
     ngx_str_t   nginxModuleResponseHeaders;
+    ngx_str_t   nginxModuleOtelExporterOtelHeaders;
```
Comment (DebajitDas) [inline instrumentation/otel-webserver-module/src/nginx/ngx_http_opentelemetry_module.h:105]: nginxModuleOtelExporterHeaders

**Review by DebajitDas — CHANGES_REQUESTED** [review]

**Review by gnm444 — COMMENTED** [review]
@kpratyus : Please approve

**Review by kpratyus — APPROVED** [review]

**Review by DebajitDas — APPROVED** [review]
LGTM

**Comment by khpeet** [comment]
hey @DebajitDas @gnm444 - Per this bug: https://github.com/open-telemetry/opentelemetry-cpp-contrib/issues/322 - It appears this commit is not working properly - is there an ETA to fix this?

**Pull request merged**